test_runner/regress/test_remote_storage.py

```diff
@@ -134,21 +133,31 @@ def test_remote_storage_backup_and_restore(
     env.pageserver.start()
 
     # Introduce failpoint in download
-    pageserver_http.configure_failpoints(("remote-storage-download-pre-rename", "return"))
+    pageserver_http.configure_failpoints(("storage-sync-list-remote-timelines", "return"))
+    env.pageserver.allowed_errors.append(
+        ".*error attaching tenant: storage-sync-list-remote-timelines",
+    )
 
+    # Attach doesn't download anything, so, this should succeed
     client.tenant_attach(tenant_id)
 
+    detail = client.tenant_status(tenant_id)
+    log.info("Tenant status with active failpoint: %s", detail)
+
     # is there a better way to assert that failpoint triggered?
     wait_until_tenant_state(pageserver_http, tenant_id, "Broken", 15)
 
     # assert cannot attach timeline that is scheduled for download
-    # FIXME implement layer download retries
+    # FIXME: we used to keep retrying the initial download on failure.
```

Comment:
Outdated since Heikkis commit

Reply:
Hmm, yeah I think so.

Reply:
Good, will clean up this test wrt on-demand download.

Reply:
This test and the comments have become quite messy. I'll take a stab at cleaning it up

Reply:
I guess we raced around this, just pushed my cleanup change and then noticed yours.